### PR TITLE
Update dependabot/fetch-metadata action metadata field

### LINKED ISSUE #12138: Update dependabot metadata field
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions

### What changes are you suggesting?

As of the last docs https://github.com/dependabot/fetch-metadata the metadata field is named `dependabot-metadata`

Replace `steps.metadata` with `steps.dependabot-metadata`

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by unrevised6419**
### Why:

Closes #12138

### What's being changed:

Change metadata field according to the docs

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@iamandrewluca 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Comment by mchammer01** [comment]
👋🏻 - I'll take a look at this but will first check with the relevant team as my knowledge in this area is very limited. Thanks for your patience 🙂

File: content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions.md [hunk 1]
```diff
@@ -184,9 +184,9 @@ jobs:
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       # The following properties are now available:
-      #  - steps.metadata.outputs.dependency-names
-      #  - steps.metadata.outputs.dependency-type
-      #  - steps.metadata.outputs.update-type      
+      #  - steps.dependabot-metadata.outputs.dependency-names
```
Comment (brrygrdn) [inline content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions.md:187]: This line refers to the `id` of the previous step on line 182, which is called `metadata` in this example. In the [dependabot/fetch-metadata README](https://github.com/dependabot/fetch-metadata/blob/main/README.md?plain=1#L94) this step is called `dependabot-metadata` which is why the two examples have diverged.

I think being consistent and using `dependabot-metadata` in both places is worthwhile, but we should also update line 182.

File: content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions.md [hunk 2]
```diff
@@ -219,7 +219,7 @@ jobs:
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Add a label for all production dependencies
-        if: ${{ steps.metadata.outputs.dependency-type == 'direct:production' }}
+        if: ${{ steps.dependabot-metadata.outputs.dependency-type == 'direct:production' }}
```
Comment (brrygrdn) [inline content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions.md:222]: Same as above, line 217 should be `id: dependabot-metadata`

File: content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions.md [hunk 3]
```diff
@@ -282,7 +282,7 @@ jobs:
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Enable auto-merge for Dependabot PRs
-        if: ${{contains(steps.metadata.outputs.dependency-names, 'my-dependency') && steps.metadata.outputs.update-type == 'version-update:semver-patch'}}
+        if: ${{contains(steps.dependabot-metadata.outputs.dependency-names, 'my-dependency') && steps.dependabot-metadata.outputs.update-type == 'version-update:semver-patch'}}
```
Comment (brrygrdn) [inline content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions.md:285]: Line 280 as well

**Review by brrygrdn — CHANGES_REQUESTED** [review]
Thanks for opening this PR, I think it's worth aligning both the project README and documentation like this 👍🏻 

We just need to ensure the `id` is updated as well.

**Comment by mchammer01** [comment]
Thanks for the technical review @brrygrdn ✨ 
@iamandrewluca 👋🏻 - can you add @brrygrdn's suggestions to your PR, so I can give it a quick check and merge it? Thank you!

**Comment by iamandrewluca** [comment]
@mchammer01 @brrygrdn updated the PR.

It seems that I configured my action wrong then
https://github.com/iamandrewluca/holiday-time/blob/e35e02a93cbef5e5dcad8efbd7965d33d203ad0a/.github/workflows/dependabot.yml#L19

It sems to work because of `!=`

**Review by brrygrdn — APPROVED** [review]
Thanks for the update, this looks good to me! :shipit:

**Review by mchammer01 — APPROVED** [review]
LGTM, thanks for applying the suggestions 🙇🏻 😃

**Pull request merged**